### PR TITLE
Alerts: show a warning/error if transformations are configured

### PR DESCRIPTION
**Pull request by ryantxu**
If transformations are configured, alerts will not behave as expected so we should not let them be configured.  This PR:

* will not let you configure Alerts if transformations exist
* shows a warning if you somehow have a transformation on a panel with an alert.

### CODE REVIEW

**Comment by ryantxu** [comment]
#18054 strikes again!

image

**Review by dprokop — APPROVED** [review]
LGTM

File: public/app/features/alerting/AlertTab.tsx [hunk 1]
```diff
@@ -127,7 +129,16 @@ export class AlertTab extends PureComponent<Props> {
...
 
   render() {
-    const { alert } = this.props.panel;
+    const { alert, transformations } = this.props.panel;
+    const hasTransformations = transformations && transformations.length;
+
+    if (!alert && hasTransformations) {
+      return (
+        <EditorTabBody heading="Alert">
+          <AlertBox severity={AppNotificationSeverity.Warning} title="Alerts are not supported with transformations" />
```
Comment (marefr) [inline public/app/features/alerting/AlertTab.tsx]:
```suggestion
          <AlertBox severity={AppNotificationSeverity.Warning} title="Transformations are not supported in alert queries" />
```

File: public/app/features/alerting/AlertTab.tsx [hunk 2]
```diff
@@ -141,6 +152,13 @@ export class AlertTab extends PureComponent<Props> {
     return (
       <EditorTabBody heading="Alert" toolbarItems={toolbarItems}>
         <>
+          {alert && hasTransformations && (
+            <AlertBox
+              severity={AppNotificationSeverity.Error}
+              title="Alerts will not work with transformations configured"
```
Comment (marefr) [inline public/app/features/alerting/AlertTab.tsx]:
```suggestion
              title="Transformations are not supported in alert queries"
```

**Review by marefr — DISMISSED** [review]
Based on how we present validation error of template variables "Template variables are not supported in alert queries" I suggests writing transformation validation error in same manner.

Besides above LGTM

**Comment by ryantxu** [comment]
@marefr -- sorry to "dismiss" your review.  I think it required you to re-ok the changes after I applied them ¯\_(ツ)_/¯

**Pull request merged**